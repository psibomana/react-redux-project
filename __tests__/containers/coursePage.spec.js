import React from 'react';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import configureStore from 'redux-mock-store'
import CoursePage from '../../src/containers/coursePage'

describe('Course Page', () => {
  let store, wrapper, initialState;

  beforeEach(() => {
    const mockStore = configureStore();
    initialState = {
      CourseReducer: {
        courses: [
          {
            id: "react-flux-building-applications",
            title: "Building Applications in React and Flux",
            watchHref: "http://www.pluralsight.com/courses/react-flux-building-applications",
            authorId: "cory-house",
            length: "5:08",
            category: "JavaScript"
          },
          {
            id: "clean-code",
            title: "Clean Code: Writing Code for Humans",
            watchHref: "http://www.pluralsight.com/courses/writing-clean-code-humans",
            authorId: "cory-house",
            length: "3:10",
            category: "Software Practices"
          }
        ]
      }
    };
    store = mockStore(initialState);
    wrapper = shallow(<CoursePage store={store} />);
  })

  it('should render as expected', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it('should map state to props', () => {
    expect(wrapper.props().courses.length)
      .toEqual(initialState.CourseReducer.courses.length)
  })
});
