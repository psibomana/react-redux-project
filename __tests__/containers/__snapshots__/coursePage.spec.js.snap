// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course Page should render as expected 1`] = `
<CoursePage
  actions={
    Object {
      "addCourse": [Function],
      "deleteCourse": [Function],
      "listCourses": [Function],
      "updateCourse": [Function],
    }
  }
  courses={
    Array [
      Object {
        "authorId": "cory-house",
        "category": "JavaScript",
        "id": "react-flux-building-applications",
        "length": "5:08",
        "title": "Building Applications in React and Flux",
        "watchHref": "http://www.pluralsight.com/courses/react-flux-building-applications",
      },
      Object {
        "authorId": "cory-house",
        "category": "Software Practices",
        "id": "clean-code",
        "length": "3:10",
        "title": "Clean Code: Writing Code for Humans",
        "watchHref": "http://www.pluralsight.com/courses/writing-clean-code-humans",
      },
    ]
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
